### PR TITLE
feat: GitHub actions CI

### PR DESCRIPTION
**Pull request by meniRoy**
**run integration tests in GitHub CI actions.**

I encountered an encryption error during the setup: `Error while encrypting the text provided to safeStorage.encryptString. Encryption is not available`. 
To address this, I've added the --password-store=basic argument to the launch parameters of @vscode/test-electron, as suggested [here](https://github.com/microsoft/vscode/issues/188059)

### CODE REVIEW

**Comment by meniRoy** [comment]
[Create GH action CI workflow](https://trello.com/c/nxy07Bln/10054-create-gh-action-ci-workflow)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+    - name: Install pnpm
+      run: npm install -g pnpm
+    - name: Install dependencies
+      run: pnpm install
+    - name: Run tests
+      run: xvfb-run -a npm test
```
Comment (sagilaufer1992) [inline .github/workflows/test.yml:23]: what is this command?

File: pnpm-lock.yaml
Comment (sagilaufer1992) [inline pnpm-lock.yaml:1]: why updating the lock file? if the package json didn't update, is there a good reason to update it for this PR?

**Review by sagilaufer1992 — COMMENTED** [review]
I wonder why to update the lock file, and I'm not familiar with the command you used for running the tests.
address these questions, and the rest LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (meniRoy) [reply .github/workflows/test.yml:23]: "In headless Linux CI machines xvfb is required to run VS Code, so if Linux is the current OS run the tests in an Xvfb enabled environment"
https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions

File: pnpm-lock.yaml
Comment (sagilaufer1992) [reply pnpm-lock.yaml:1]: it seems like the installation of pnpm installed a different version than before. 
you can see in our main repo, we expect to use a specific version of pnpm, specified in the package json under "engines". 
make the GH action install a SPECIFIC pnpm version, that still corresponds to this lockfile, so we won't fave such issues later on in another pnpm upgrade

Comment (meniRoy) [reply pnpm-lock.yaml:1]: eb88a6f

Comment (sagilaufer1992) [inline pnpm-lock.yaml:1]: can we revert the pnpm lock? or is it required now that we have pnpm as an engine?

**Review by sagilaufer1992 — APPROVED** [review]
I approve, but please, if the pnpm need to be a dependency, add it, and if the lock file doesn't need to be changed, revert it

File: pnpm-lock.yaml
Comment (meniRoy) [reply pnpm-lock.yaml:1]: 652cbe5

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "description": "",
   "version": "0.0.1",
   "engines": {
-    "vscode": "^1.71.0"
+    "vscode": "^1.71.0",
+    "pnpm": "6.35.1"
```
[the remark this answers is not in the document]
Comment (meniRoy) [reply package.json:9]: pnpm is not a dev dependency its installed globally with fixed version [here](https://github.com/env0/vscode-env0/blob/652cbe57f6a51887d3b1e1a3e2f59a7655d09506/.github/workflows/test.yml#L19)

**Pull request merged**